### PR TITLE
Fix remote and realm join

### PR DESCRIPTION
**Pull request by Yehonal**
This fixes connection when server is not in same host of wowser

removing localhost hardcoded inside and implementing port configuration.

Also sending correct opcode when realm is not 0 

app.writeUnsignedInt(this.realm.id);

### CODE REVIEW

File: src/lib/game/handler.js [hunk 1]
```diff
@@ -27,9 +27,10 @@ class GameHandler extends Socket {
   }
 
   // Connects to given host through given port
-  connect(host, port) {
+  connect(host, realm) {
+    this.realm=realm;
```
Comment (timkurvers) [inline src/lib/game/handler.js]: Could you add spaces around operators, adhering to the [styleguide](http://eslint.org/docs/rules/space-infix-ops)?

File: src/lib/game/handler.js [hunk 2]
```diff
@@ -27,9 +27,10 @@ class GameHandler extends Socket {
   }
 
   // Connects to given host through given port
```
Comment (timkurvers) [inline src/lib/game/handler.js]: Could you update this comment to reflect the change in method signature?

File: src/components/realms/index.jsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ class RealmsScreen extends React.Component {
   }
 
   connect(realm) {
-    session.game.connect('localhost', realm.port);
+    session.game.connect(session.authstate.host, realm);
```
Comment (timkurvers) [inline src/components/realms/index.jsx]: Is this state necessary? Wouldn't `session.auth.host` suffice?

File: src/components/realms/index.jsx [hunk 4]
```diff
@@ -44,7 +44,11 @@ class RealmsScreen extends React.Component {
   }
 
   _onRealmSelect(event) {
-    this.setState({ realm: event.target.value });
+    const realms = session.realms.list;
+    this.setState({
+        realm: realms[event.target.selectedIndex],
+        realms: realms
```
Comment (timkurvers) [inline src/components/realms/index.jsx]: Where is `realms` used?

File: src/lib/game/handler.js (hunk 1 above)
Comment (Yehonal) [inline src/lib/game/handler.js]: I've fixed it, however i suggest you to make a "confirm and squash" before accept this PR

https://help.github.com/articles/about-pull-request-merge-squashing/

**Comment by Yehonal** [comment]
I've cleaned it 

However about auth state, i don't remember why i needed to create that state... but there are only 2 possibilities:

1) i've not noticed the session.auth.host

2) session.auth.host is not the same of component state ( value inserted into text input ) . I've checked it reading the code, but seems that are the same values. I cannot test it now

However i've cleaned all the code and i think you can squash and merge if you want.

**Comment by timkurvers** [comment]
Had you been able to test whether the `setState` changes are required?

**Comment by fallenoak** [comment]
Since Wowser currently only runs in the browser, its ability to connect to a given host and port via TCP is limited to what's exposed by a Websockify proxy. See here: https://github.com/wowserhq/wowser#socket-proxies

While supporting arbitrary hostname and port combinations is something we'd like to do, I'm not really sure there's much to be gained by adding this before Wowser is either capable of running in another context (like an Electron-ified desktop app), or is capable of specifying configuration options for the proxy server.

**Comment by timkurvers** [comment]
Great catch @fallenoak, had completely forgotten about that!

**Comment by Yehonal** [comment]
I'm using it with your proxy configured to handle this connection. I've
tested in any kind of configurations: wowser and proxy on same host of
server in a local context , wowser and proxy in remote on same host of
server, wowser and proxy on remote or on local with server in another host.
With chrome or firefox I've any kind of problem. This configuration is
needed to allow to host wowser on remote and allowing people to connect
without any proxy configured..with a small trick on server proxy.

If you explain me where is exaxtly the problem i can test it better

Il 01/Ago/2016 14:42, "Tim Kurvers" <notifications@github.com> ha scritto:

**Comment by fallenoak** [comment]
@Yehonal Right, but if the auth server is returning some specific port number, and that port is actually running a Websockify proxy, how is the real World of Warcraft client still able to connect?

**Pull request closed without merging**